docs/developer/decision-records/2023-02-28-processing-callbacks/README.md

```diff
@@ -0,0 +1,147 @@
+# Processing Callbacks
+
+## Decision
+
+Implement the ability for clients to specify callback endpoints to be invoked at specific points during request processing:
+
+- Callback endpoints can be specified for _Contract Negotiation_ and _Transfer Process_ state transitions when a request is made via the _Management API_
+- Callback endpoints are scoped to the particular state machine instance associated with the request.
+- Clients can request the callbacks to be invoked *transactionally* or on a *best-effort* basis.
```

Comment:
Is it a good thing to have transactional hooks that rely on external services? Which kind of use cases would they cover?

Reply:
This is needed for the case where delivery needs to be guaranteed. Some of these services need not be external (the endpoint could be in process), but even with external services, the latency would not be an issue. Also, the default behavior is non-transactional.